src/middlewares/restrict-user-details.js

```diff
@@ -0,0 +1,102 @@
+module.exports = (config, { strapi }) => {
+  return async (ctx, next) => {
+    await next();
+    try {
+      if (ctx.request.method === "GET" && ctx.request.url.includes("/api/")) {
+        const handler = ctx.state.route.handler;
+        const user = ctx.state.user;
+        const idx = handler.lastIndexOf(".");
+        const modelName = handler.substring(0, idx);
+        const action = handler.substring(idx + 1);
+        const metadata = strapi.db.metadata;
```

Comment:
What are examples of metadata?

Reply:
Metadata is a Map-alike object where key is a modelName and value is MetaModel, which contains almost all the same info you can find under `content-types/schema.json` See https://github.com/strapi/strapi/blob/develop/packages/core/database/src/types/index.ts#L216 and https://github.com/strapi/strapi/blob/develop/packages/core/database/src/metadata/metadata.ts